src/gif.rs

```diff
@@ -224,7 +269,10 @@ impl<W: Write> Encoder<W> {
         result
     }
     /// Encodes Frames.
-    pub fn encode_frames(&mut self, frames: animation::Frames) -> ImageResult<()> {
+    pub fn encode_frames<Frames>(&mut self, frames: Frames) -> ImageResult<()>
+    where 
+        Frames: Iterator<Item = animation::Frame>
```

Comment:
I'd recommend `IntoIterator<animation::Frame>`, that makes this api more strictly more broad.

It strikes me as odd that I can not use `animation::Frames` as an argument to `encode_frames`. Generic argument should probably be named different, and maybe a `try_encode_frames` which takes `IntoIterator<ImageResult<animation::Frame>>`?

Minor stylistic choice: Rewriting the `for`-body in terms of `map` and `collect`, avoiding the `if let Err(_)` control flow at the end and `Ok(())`.

Reply:
First of all, thank you for the review!

Yes, using `IntoIterator` instead of `Iterator` is a great idea, I will certainly do that. Also `try_encode_frames` seems like it would have some nice use cases so I'll make that as well, having it return the error in case of an `Err`.

On the other hand I don't really see how `if let Err(_)` could be alleviated. Could you elaborate on that?

Reply:
Sure, the body could be written as:

```
frames.into_iter().map(|frame| {
    // get the delay before coverting img_frame
    let frame_delay = img_frame.delay().to_integer();
    // convert img_frame into RgbaImage
    let rbga_frame = img_frame.into_buffer();

    // Create the gif::Frame from the animation::Frame
    let mut frame = Frame::from_rgba(rbga_frame.width() as u16, rbga_frame.height() as u16, &mut rbga_frame.into_raw());
    frame.delay = frame_delay;

    // encode the gif::Frame
    self.encode(&frame)
}).collect()
```

It's mostly a matter of style and personal taste though.

Reply:
Hmm but in this case `collect` would resolve to `collect<ImageResult<()>>`. As long as I'm not missing something I think this would go against the general expectation of what `collect` means and what `ImageResult` means so I prefer the original form.

Reply:
Depends on what is general expectation of `collect`. From its own documentation:

```
This is one of the more powerful methods in the standard library, used in a variety of contexts.
[…]
One of the keys to collect()'s power is that many things you might not think of as 'collections' actually are. […] And a collection of Result<T, E> can be thought of as single Result<Collection<T>, E>. See the examples below for more.
```